File: M/1189-maximum-number-of-balloons/main.go
package main

import "fmt"

func maxNumberOfBalloons(text string) int {
	buf := make([]int, 5)
	for _, b := range text {
		if b == 'b' {
			buf[0]++
		} else if b == 'a' {
			buf[1]++
		} else if b == 'l' {
			buf[2]++
		} else if b == 'o' {
			buf[3]++
		} else if b == 'n' {
			buf[4]++
		}
	}
	buf[2] /= 2
	buf[3] /= 2
	min := 1<<31 - 1
	for _, c := range buf {
		if min > c {
			min = c
		}
	}
	return min
}

func main() {
	fmt.Println("a")
}

File: M/1189-maximum-number-of-balloons/main.rs
impl Solution {
  pub fn max_number_of_balloons(text: String) -> i32 {
    let mut buf: (i32, i32, i32, i32, i32) = (0, 0, 0, 0, 0);
    text.as_bytes().iter().for_each(|&b| match b {
      b'a' => buf.0 += 1,
      b'b' => buf.1 += 1,
      b'l' => buf.2 += 1,
      b'n' => buf.3 += 1,
      b'o' => buf.4 += 1,
      _ => {}
    });

    *[buf.0, buf.1, buf.2 / 2, buf.3, buf.4 / 2].iter().min().unwrap()
  }
}
